Add French translation for OptiGUI 3

JustArthur opened the pull request.

JustArthur committed:
Add French translation

opekope2 commented:
Hello!
Thank you for your patience — I was working on updating the translations, which took more time than I expected. I updated the translations on the `dev` branch with over 70 new translations.
If you want to submit updated translations for the next OptiGUI 3 alpha release, you have at least 30 days.

JustArthur committed:
Add new translations keys

JustArthur committed:
Merge branch 'opekope2:dev' into dev

JustArthur committed:
Merge branch 'opekope2:dev' into dev

opekope2 commented:
After implementing more features, I had to make some minor changes to the translations.

After skimming through the translation file, I have some suggestions:

* `optigui.rp_converter.convert_properties`: OptiFine uses the `.properties` extension, but it can stay like how it is, I don't speak French
* For `optigui.rp_loader.warn.no_interaction_target`, I didn't provide enough context. `blocks`, `entities`, `items`, `inventory`, and `unknown` will be part of V2 JSON resources (just like `texture.PATH` in OptiFine properties):
  ```json
  {
    "blocks": [],
    "entities": [],
    "items": [],
    "inventory": false,
    "unknown": true
  }
  ```

JustArthur committed:
Merge branch 'opekope2:dev' into dev

JustArthur committed:
Update fr_fr

JustArthur commented:
Thanks for the clarification!
I’ve updated the French translation accordingly:

optigui.rp_converter.convert_properties → changed to

“Convertir les fichiers ".properties" d'OptiFine”
to better reflect the fact that OptiFine specifically uses the .properties file extension.

optigui.rp_loader.warn.no_interaction_target → reworded to match the context you mentioned for the upcoming V2 JSON resources.
It now reads as:

“Aucune cible d’interaction (blocs, entités, objets, inventaire ou inconnu) n’a été configurée dans la ressource JSON”
which better aligns with the new blocks, entities, items, inventory, and unknown fields planned for V2.

I’ve also translated the new strings and made minor adjustments to some existing lines to keep the French localization consistent and accurate.

opekope2 commented:
The translation keys for OptiGUI 3 alpha 4 are **finished**, no more additions, maybe small clarifications.
I organized the translation keys since your last revision, so some of them may be out of place. Other than that, 1 translation is missing.
You have at least 30 days to update your translation.

JustArthur committed:
Merge branch 'opekope2:dev' into dev

JustArthur committed:
Update and refine French translation for OptiGUI

Updated the existing French localization with corrected phrasing, consistent terminology, and full coverage of missing OptiGUI strings. Improved accuracy across inspector, resource loader, and converter modules.

opekope2 commented:
I reviewed your translation, and have a couple of suggestions:
*  I changed INI translations to `.ini`, similar to `.properties`
* `optigui.validation.error.not_a_relative_identifier`: relative resource location refers to an OptiGUI-specific variant of https://minecraft.wiki/w/Resource_location, so `./texture.png` can be used instead of `namespace:path/to/texture.png`
* `optigui.validation.error.empty_map`: map refers to [this data structure](https://en.wikipedia.org/wiki/Associative_array), not a map item
